### ISSUE TITLE
Error using typing_extensions.Protocol with Python 3.6

### ISSUE THREAD

**Issue by tsibley** [microsoft/pyright#2987]
**Describe the bug**
pyright 1.1.217 (installed from NPM) incorrectly reports `Use of "Protocol" requires Python 3.7 or newer` when analyzing for Python 3.6 and importing `Protocol` from `typing_extensions`.

**To Reproduce**
With the code below saved in `example.py`, run:

```console
$ pyright --pythonversion 3.6 example.py
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/pyright-repro/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/pyright-repro/example.py
  /tmp/pyright-repro/example.py:3:9 - error: Use of "Protocol" requires Python 3.7 or newer
1 error, 0 warnings, 0 informations
Completed in 0.597sec

$ pyright --version
pyright 1.1.217
```

**Expected behavior**
importing `Protocol` from `typing_extensions` should not trigger an error.

**Screenshots or Code**
```python
from typing_extensions import Protocol

class A(Protocol):
    pass
```

**VS Code extension or command-line**
pyright 1.1.217 from NPM.

**Additional context**
Regression of a previously fixed bug: https://github.com/microsoft/pyright/issues/763

**Comment by erictraut** [comment]
The problem here is that `typing_extensions.pyi` was recently changed to import the symbol `Protocol` from `typing.pyi` rather than defining its own version of the symbol. Now pyright can't tell the difference between the two because they are the same symbol. If a Python 3.6 source file imports the symbol from `typing`, it will result in a runtime exception, which is why the error message was added originally.

I'll need to ask the maintainers of typeshed why they made this change and see if they're willing to revert it.

**Comment by erictraut** [comment]
I submitted the change to the typeshed repo. It has been merged, and I updated pyright's copy of the typeshed stubs accordingly, so this will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.219, which I just published. It will also be in the next release of pylance.
